### ISSUE TITLE
Reference information is lost for editable packages

### ISSUE THREAD

**Issue by ezaiakinsc** [conan-io/conan#5565]
- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

While executing _conan install_ with editable packages, information from following attributes: **version**, **user**, **channel** is lost from conanfile object, and is not accessible at least in the _requirements_() method of the editable conanfile.

Of course, version attribute is lost only if it was set by _conan export_ command, not by "version = xxx" in conanfile, however, inaccessibility of that data prevents using highly-interconnected libraries, for example like this:

```
class LibraryAFromMonorepo(ConanFile):
    def requirements(self):
        self.requires(f"LibraryBFromMonorepo/{ self.version }@{ self.user }/{ self.channel }")
```

- which should perfectly usable like this:
conan export {path_to_lib_a_in_monirepo} LibraryAFromMonorepo/{repo_git_hash}@{team}/{branch}.
conan export {path_to_lib_b_in_monirepo} LibraryBFromMonorepo/{repo_git_hash}@{team}/{branch}.

That code perfectly works in "cache mode", and official documentation explicitly states similar use cases as valid. But same code gets only "None" in editable mode of the package during _conan install_, even with full reference specification is known in advance.

It looks reasonable, since Conan knows those values from editable reference specification, provide them in the same manner as in cache mode, for consistency. Right now this bug prevents any usage of calculated packages references, on any Conan version.

If that "None"-behavior is intentional for another feature - can values from "editable add" reference to be preserved in some backup fields?

**Comment by jgsogo** [comment]
Hi, @ezaiakinsc , can you provide a minimun reproducible example? I'm trying with something like this:

```python
from conans import ConanFile

class LibConan(ConanFile):
    name = "lib"

    def requirements(self):
        self.output.info("*"*20)
        self.output.info(self.version)
        self.output.info(self.user)
        self.output.info(self.channel)
```

```bash
conan create . lib/0.1@issue/testing
conan editable add . lib/0.1@issue/testing
```
And I'm able to consume that package as a requirement and I get the values of `version`, `user` and `channel` properly.

I'm sure I'm missing something and up to my knowledge having `None` is not intended, so it is probably a bug in the editable feature, but I need to reproduce iit.

Thanks

**Comment by ezaiakinsc** [comment]
Hi @jgsogo, did you tried to do _conan install_ after this editable application? 
Here is my example (similar to yours):
```
class ProxyLib(ConanFile):
    name = "ProxyLib"
    generators = ("nlo_cmake_link", "json")
    exports_sources = ["CMakeLists.txt", "src/*", "android/*", "nlo-android/*"]
    options = {"isDemo": [False, True]}
    default_options = {"isDemo": False}

    def requirements(self):
        self.requires(f"{self.name}/{self.version}@{self.user}/{self.channel}")
```
and after this
```
conan editable add ~/Repos/NloSanbox/ProxyLib ProxyLib/1.0@snap/stable
```
We are trying to do corresponding _conan install_ of the package that uses this ProxyLib. Bug might look different on different Conan version, on 1.18 output is this:
```
ERROR: conanfile.py (ProxyLib/None): Error in requirements() method, line 12
        self.requires(f"{self.name}/{self.version}@{self.user}/{self.channel}")
        ConanException: CONAN_USERNAME environment variable not defined, but self.user is used in conanfile
```

**Comment by jgsogo** [comment]
Yes, I'm consuming and installing a library that depends on the editable one, and I'm seeing those lines printed with the proper values. Also checked that I'm running `1.18`.

Can you paste the list of commands you are running and the two `conanfile.py` you are using (or two that are enough to reproduce the issue) so I can work on them to reproduce the issue? Thanks

**Comment by ezaiakinsc** [comment]
While isolating the issue, I realized it is our mistake in _conan install_ command: the one you see mistakenly points to the root of editable ProxyLib, and thus the output. 

Clearly, if conanfile without version etc is used as conan install root, it will result None as version etc, so what we encountered is not a bug of conan, but our incorrect invocation.

Very sorry for wasting your time, closing the issue.
